addons/mrp/models/mrp_production.py

```diff
@@ -1380,8 +1380,9 @@ def _get_name_backorder(self, name, sequence):
         if not sequence:
             return name
         seq_back = "-" + "0" * (SIZE_BACK_ORDER_NUMERING - 1 - int(math.log10(sequence))) + str(sequence)
-        if re.search("-\\d{%d}$" % SIZE_BACK_ORDER_NUMERING, name):
-            return name[:-SIZE_BACK_ORDER_NUMERING-1] + seq_back
+        regex = r"-\d+$"
+        if re.search(regex, name):
+            return re.sub(regex, seq_back, name)
```

Comment:
Nice and cleaner, Thank you.

Small suggestion maybe, it use a compiled regex instead (but it is a detail):
```python
regex = re.compile(r"-\d+$")
if regex.search(name):
    return regex.sub(seq_back, name)
```